### ISSUE TITLE
error: unknown option: --watch for component-tests after updating nx to 18.0.4

### ISSUE THREAD

**Issue by tar-aldev** [nrwl/nx#21984]
### Current Behavior

When trying to run nx component-test <project> --watch it fails with 
```
  error: unknown option: --watch
```

I can reproduce it both on new and existing project. As a result it is impossible to run cypress tests in watch mode.

### Expected Behavior

--watch flag should work for tests.
It is also not possible to set it in project.json targets as an option.

### Steps to Reproduce

1. create cypresss ocmponent tests
2. try to run it with --watch

### Nx Report

```shell
Node   : 18.18.2
   OS     : darwin-arm64
   pnpm   : 8.7.6
   
   nx                 : 18.0.4
   @nx/js             : 18.0.4
   @nx/jest           : 18.0.4
   @nx/linter         : 18.0.4
   @nx/eslint         : 18.0.4
   @nx/workspace      : 18.0.4
   @nx/cypress        : 18.0.4
   @nx/devkit         : 18.0.4
   @nx/eslint-plugin  : 18.0.4
   @nx/express        : 18.0.4
   @nx/next           : 18.0.4
   @nx/node           : 18.0.4
   @nx/playwright     : 18.0.4
   @nx/react          : 18.0.4
   @nx/storybook      : 18.0.4
   @nrwl/tao          : 18.0.4
   @nx/web            : 18.0.4
   @nx/webpack        : 18.0.4
   typescript         : 5.3.3
   ---------------------------------------
   Community plugins:
   @ago-dev/nx-aws-cdk-v2 : 1.6.1
```

### Operating System

- [ ] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information
If I put this into `project.json`
```
"component-test": {
      "executor": "@nx/cypress:cypress",
      "options": {
        "cypressConfig": "...",
        "skipServe": true,
        "testingType": "component"
      },
      "configurations": {
        "production": {
          "devServerTarget": "..."
        }
      }
    }
```
then I am able to pass `--watch` flag. Not sure if this is an expected bahviour or not

**Comment by DevNebulae** [comment]
In general I have noticed that Cypress doesn't work too well with inferred tasks / Project Crystal. I still need to add all the manual configuration to make either work, even for newly generated projects. Issues ranging from certain command (i.e. `--watch`) not working to certain options not being available that are described in the documentation.

**Comment by andrewalderson** [comment]
Just ran into the same issue with e2e tests. As above, I had to specifically add the target to project.json to make it work.

**Comment by McGiogen** [comment]
Same here. Probably `--watch` and other flags doesn't works because infer targets are using `nx:run-commands` instead of `@nx/cypress:cypress` as executor and they are custom flags of `@nx/cypress` plugin.

This is our infer e2e target:

```json
    "e2e": {
      "cache": true,
      "inputs": [
        "default",
        "^production"
      ],
      "outputs": [
        "{workspaceRoot}\\dist\\cypress\\apps\\evohrp-e2e\\videos",
        "{workspaceRoot}\\dist\\cypress\\apps\\evohrp-e2e\\screenshots"
      ],
      "executor": "nx:run-commands",
      "options": {
        "cwd": "apps/evohrp-e2e",
        "command": "cypress run"
      },
      "configurations": {
        "production": {
          "command": "cypress run --env webServerCommand=\"nx run evohrp:serve:production\""
        }
      }
    },
```

You can check yourself with command `nx show project <project-name>`

**Comment by leosvelperez** [comment]
Thanks for reporting this!

We addressed running the e2e tests in watch mode in https://github.com/nrwl/nx/pull/22556 that was released in [Nx 18.2.0](https://github.com/nrwl/nx/releases/tag/18.2.0). The Cypress plugin will now [infer an `open-cypress` task](https://nx.dev/nx-api/cypress/documents/overview#watching-for-changes-headed-mode) that runs `cypress open`.

I'll take a look for Component Testing.

**Comment by leosvelperez** [comment]
@tar-aldev could you update your repo to at least Nx 18.2.0 and try running the newly inferred `open-cypress` task?
